### DISCUSSION TITLE
"/godata/db/h2db/cruise.mv.db" is locked

### DISCUSSION THREAD

**Discussion by elipesatya** [gocd/gocd#12716] [Development]
Hi All 

I'm dockerizing our static/standalone GoCD server.

Hence copied the content from the static GoCD server to EFS and mounted that EFS content into the container (GoCD server container).

And with that, container starts and runs but it can't fetch any data, and I see the below in the log:

```jvm 2    | Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database may be already in use: null. Possible solutions: close all other connection(s); use the server mode [90020-200]

jvm 2    | Caused by: java.lang.IllegalStateException: The file is locked: nio:/godata/db/h2db/cruise.mv.db [1.4.200/7]
```

Looks like the static/old server acquired the lock and when the data from that server copied to EFS lock is still maintained and hence the new server has the db file locked and couldn't fetch any data. 

Wonder, how are the people handling this situation ?
We need the old server running as well when the new server is being brought up within the docker container. 

Any inputs will be of great help. 

Many thanks, 
Satya

**Comment by chadlwilson** [comment]
If you copied the database file from the old server to a new file in a separate EFS volume, I don't think the filesystem lock should/would be retained, as it represents a different file.

Are you sure you don't have two GoCD containers running, trying to mount the same database/godata dir? Do you have a stopped but not `rm`ed container? It's definitely not safe to have two GoCD servers or containers trying to share any storage (database, config, artifacts etc)

**Comment by elipesatya** [comment]
Thank you Chad.

Yes, copying into a new file could help bringing up another server (container) alongside the existing one.

We have saved all our content into EFS under /godata and maps that into the container as /godata.

We are using gocd/gocd-server:v22.3.0.
It all runs good, mapping was fine too but just one thing that’s not happening is “.ssh” folder.

I have .ssh with all required keys in EFS under /godata and /godata within the container also has .ssh but not /go-working-dir.

Is that supported, am I mis-configuring it, or do we need to handle that outside of the base image ?

Many thanks

**Reply by chadlwilson** [reply]
https://groups.google.com/g/go-cd/c/1cfo3ILKYIE

**Comment by elipesatya** [comment]
Will send this as a separate thread on email groups too.

**Reply by chadlwilson** [reply]
Would be good to not post the same thing in multiple places/threads/discussions. There are more people on the mailing list, but it's much easier to format and reply here, and more searchable :-)
